backend/s3/s3.go

```diff
@@ -2987,7 +2987,9 @@ func (o *Object) Open(ctx context.Context, options ...fs.OpenOption) (in io.Read
 	if resp.LastModified == nil {
 		fs.Logf(o, "Failed to read last modified: %v", err)
 	}
-	o.setMetaData(resp.ETag, resp.ContentLength, resp.LastModified, resp.Metadata, resp.ContentType, resp.StorageClass)
```

Comment:
I don't think this fix is right - it will get the length wrong when o.fs.opt.NoHeadObject in used.

The correct length will be returned in the Content-Range header `Content-Range: bytes 0-0/6` which is returned in `resp.ContentRange`.

So something like this (untested)

```go
	// read size from ContentLength or ContentRange
	size := resp.ContentLength
	if resp.ContentRange != nil {
		var contentRange = *resp.ContentRange
		slash := strings.IndexRune(contentRange, '/')
		if slash >= 0 {
			i, err := strconv.ParseInt(contentRange[slash:], 10, 64)
			if err == nil {
				size = &i
			} else {
				fs.Debugf(o, "Failed to find parse integer from in %q: %v", contentRange, err)
			}
		} else {
			fs.Debugf(o, "Failed to find length in %q", contentRange)
		}
	}
	o.setMetaData(resp.ETag, size, resp.LastModified, resp.Metadata, resp.ContentType, resp.StorageClass)
```